lib/image_compare/modes/base.rb

```diff
@@ -6,11 +6,11 @@ class Base
       require "image_compare/rectangle"
       include ColorMethods
 
-      attr_reader :result, :threshold, :lower_threshold, :different_areas, :exclude_rect, :include_rect
+      attr_reader :result, :threshold, :lower_threshold, :different_areas, :exclude_rects, :include_rect
 
-      def initialize(threshold: 0.0, lower_threshold: 0.0, exclude_rect: nil, include_rect: nil)
+      def initialize(threshold: 0.0, lower_threshold: 0.0, exclude_rects: [], include_rect: nil)
         @include_rect = Rectangle.new(*include_rect) unless include_rect.nil?
-        @exclude_rect = Rectangle.new(*exclude_rect) unless exclude_rect.nil?
+        @exclude_rects = @exclude_rects = exclude_rects.empty? ? Set.new : Set.new(exclude_rects.map { |rect| Rectangle.new(*rect) })
```

Comment:
Please consider changing to:
```
@exclude_rects = exclude_rects.empty? ? Set.new : Set.new(exclude_rects.map { |rect| Rectangle.new(*rect) })
```

_**Don't forget to fix the error on line 17 too**_